tests/core/debug/task.yaml

```diff
@@ -0,0 +1,42 @@
+summary: Check that UC debug settings work
+
+details: |
+  Check that UC debug settings enable logging for snapd and
+  setting the log level for systemd as expected.
+
+execute: |
+  if [ "$SPREAD_REBOOT" = 0 ]; then
+      # Remove the variable introduced by prepare.sh
+      sed -i 's/SNAPD_DEBUG=1//' /etc/systemd/system/snapd.service.d/local.conf
+      systemctl daemon-reload
+      systemctl restart snapd
+
+      # Check enable/disable snapd debug traces
+      snap set system debug.snapd.log=true
+      MATCH SNAPD_DEBUG=1 < /var/lib/snapd/environment/snapd.conf
+      systemctl restart snapd
+      MATCH SNAPD_DEBUG=1 < /proc/"$(pgrep snapd)"/environ
+      snap set system debug.snapd.log=false
+      not test -f /var/lib/snapd/environment/snapd.conf
+      systemctl restart snapd
+      not MATCH SNAPD_DEBUG=1 < /proc/"$(pgrep snapd)"/environ
+
+      # Check set systemd log level
+      log_level=$(systemctl log-level)
+      test "$log_level" = info
+      snap set system debug.systemd.log-level=debug
+      MATCH LogLevel=debug < /etc/systemd/system.conf.d/20-debug_systemd_log-level.conf
```

Comment:
You could also use `systemd-analyze cat-config systemd/system.conf` to get the whole config with all the drop-ins. But there is already lots of test.

Reply:
Right, but maybe the way it is now is more targetd